frontend/src/app/components/ExperimentPage.js

```diff
@@ -169,12 +178,18 @@ export default class ExperimentPage extends React.Component {
                 <h1 data-hook="title">{title}</h1>
                 <h4 data-hook="subtitle" className="subtitle">{subtitle}</h4>
               </header>
-              {hasAddon && <div className="experiment-controls" data-hook="active-user">
+              {hasAddon && validVersion && <div className="experiment-controls" data-hook="active-user">
                 {!enabled && <a onClick={e => this.highlightPrivacy(e)} data-hook="highlight-privacy" className="highlight-privacy" data-l10n-id="highlightPrivacy">Your privacy</a>}
                 {enabled && <a onClick={e => this.feedback(e)} data-l10n-id="giveFeedback" data-hook="feedback" id="feedback-button" className="button default" target="_blank" href={surveyURL}>Give Feedback</a>}
                 {enabled && <button onClick={e => this.renderUninstallSurvey(e)} style={{ width: progressButtonWidth }} data-hook="uninstall-experiment" id="uninstall-button" className={classnames(['button', 'secondary'], { 'state-change': isDisabling })}><span className="state-change-inner"></span><span data-l10n-id="disableExperimentTransition" className="transition-text">Disabling...</span><span data-l10n-id="disableExperiment" data-l10n-args={JSON.stringify({ title })} className="default-text">Disable <span data-hook="title">{title}</span></span></button>}
                 {!enabled && <button onClick={e => this.installExperiment(e)} style={{ width: progressButtonWidth }} data-hook="install-experiment" id="install-button"  className={classnames(['button', 'default'], { 'state-change': isEnabling })}><span className="state-change-inner"></span><span data-l10n-id="enableExperimentTransition" className="transition-text">Enabling...</span><span data-l10n-id="enableExperiment" data-l10n-args={JSON.stringify({ title })} className="default-text">Enable <span data-hook="title">{title}</span></span></button>}
               </div>}
+              {hasAddon && !validVersion &&
```

Comment:
Let's separate these logical chunks into separate methods, e.g.

```javascript
renderMinimumVersionNotice() {
  if (hasAddon && !this.isValidVersion() {
    return (
      <div className="upgrade-notice">...</div>
    );
  }
  return null;
}
```

Then just call them here: `{this.renderMinimumVersionNotice()}`